### PR TITLE
Added parsing of skip offset for VAST ad

### PR DESCRIPTION
**Pull request by balavor**
## Changes
- Added parsing for skip offset. It can be presented as a percentage or as an exact time.
- To parse the time in seconds from VAST I've added a method for this specific case

@VerizonAdPlatforms/video-partner-sdk-developers: Please review.

### CODE REVIEW

File: sources/advertisements/VASTParser.swift [hunk 1]
```diff
@@ -491,6 +504,21 @@ enum VASTParser {
...
+        let components = string.components(separatedBy: ":")
+        guard components.count == 3 else { return nil }
+        return components
+            .compactMap { return Double($0) }
+            .enumerated()
+            .map {
+                let multiplier = pow(60, abs($0.offset - 2)) as NSNumber
+                return $0.element * multiplier.doubleValue
+            }
+            .reduce(0.0) { $0 + $1 }
```
Comment (VladyslavAnokhin) [inline sources/advertisements/VASTParser.swift]: reduce(0.0, +)

File: sources/advertisements/VASTParser.swift [hunk 2]
```diff
@@ -491,6 +504,21 @@ enum VASTParser {
         
         return delegateStack
     }
+    
+    static func parseTime(from string: String) -> Double? {
+        let components = string.components(separatedBy: ":")
+        guard components.count == 3 else { return nil }
+        return components
+            .compactMap { return Double($0) }
```
Comment (VladyslavAnokhin) [inline sources/advertisements/VASTParser.swift]: compactMap(Double.init)

File: sources/advertisements/VASTParser.swift [hunk 3]
```diff
@@ -491,6 +504,21 @@ enum VASTParser {
...
+            .enumerated()
+            .map {
+                let multiplier = pow(60, abs($0.offset - 2)) as NSNumber
```
Comment (VladyslavAnokhin) [inline sources/advertisements/VASTParser.swift]: Magic :)

File: sources/advertisements/VASTParser.swift [hunk 4]
```diff
@@ -491,6 +504,21 @@ enum VASTParser {
...
     }
+    
+    static func parseTime(from string: String) -> Double? {
+        let components = string.components(separatedBy: ":")
+        guard components.count == 3 else { return nil }
+        return components
+            .compactMap(Double.init)
+            .enumerated()
+            .map {
+                let multiplier = pow(60, abs($0.offset - 2)) as NSNumber
```
Comment (AndriiMoskvin) [inline sources/advertisements/VASTParser.swift]: Why so complex? Please explain how this works.

**Review by AndriiMoskvin — DISMISSED** [review]

File: sources/advertisements/VASTParser.swift (hunk 4 above)
Comment (balavor) [reply sources/advertisements/VASTParser.swift]: It is much shorter than if I write it as `components[0] * 3600` and so on.

**Comment by balavor** [comment]
@AndriiMoskvin @VladyslavAnokhin I've added logic to check date format that we get in VAST.

File: sources/advertisements/VASTParser.swift [hunk 5]
```diff
@@ -491,6 +506,27 @@ enum VASTParser {
...
+    static func parseTime(from string: String) -> Double? {
+        let components = string.components(separatedBy: ":")
+        guard components.count == 3 else { return nil }
+        let secondsArray: [Double] = components
+            .compactMap(Double.init)
+            .enumerated()
+            .compactMap {
+                let multipliers: [Double] = [3600, 60, 1]
+                let maxValues: [Double] = [24, 60, 60]
+                let floatingValues = [false, false, true]
```
Comment (VladyslavAnokhin) [inline sources/advertisements/VASTParser.swift]: Looks like this is new data struct. You can create `strcut` with all this rules  for each time component and put it into the array

**Review by AndriiMoskvin — DISMISSED** [review]
Check my solution in Slack.

File: sources/advertisements/VASTParser.swift [hunk 6]
```diff
@@ -491,6 +506,54 @@ enum VASTParser {
...
         return delegateStack
     }
+    
+    struct VASTTime {
+        enum Multipliers: Int {
+            case hours = 3600
+            case minutes = 60
+        }
+        
+        struct MaxValues {
```
Comment (VladyslavAnokhin) [inline sources/advertisements/VASTParser.swift]: struct -> enum

**Review by VladyslavAnokhin — DISMISSED** [review]
👍 nice enum :)

File: sources/advertisements/VASTParser.swift [hunk 7]
```diff
@@ -491,6 +506,61 @@ enum VASTParser {
...
+            private var stringTime: String {
+                switch self {
+                case .hours(let value): return value[self.index]
+                case .minutes(let value): return value[self.index]
+                case .seconds(let value): return value[self.index]
+                }
+            }
+            
+            var resultInSeconds: Int? {
+                guard let result = Double(self.stringTime)?.rounded() else { return nil }
```
Comment (VladyslavAnokhin) [inline sources/advertisements/VASTParser.swift]: `Int(Double(self.stringTime)?.rounded())`

**Review by VladyslavAnokhin — APPROVED** [review]

**Review by AndriiMoskvin — APPROVED** [review]

**Pull request merged**